component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSIdentityStorageImpl.java

```diff
@@ -287,6 +290,15 @@ private void mapToProfileEntity(Profile profile, IdentityEntity identityEntity)
 
       } else if (!Profile.EXPERIENCES_SKILLS.equals(profileProperty.getKey())) {
         Object val = profileProperty.getValue();
+        if ((Profile.FIRST_NAME.equalsIgnoreCase(profileProperty.getKey())
+            || Profile.LAST_NAME.equalsIgnoreCase(profileProperty.getKey()))
+            && !profileProperty.getValue().equals(entityProperties.get(profileProperty.getKey()))) {
```

Comment:
Must be sure that avatar exists before calling `getAvatarFile` which will generate a new File if not existing.

```suggestion
            && !profileProperty.getValue().equals(entityProperties.get(profileProperty.getKey()))
            && identityEntity.getAvatarFileId() != null) {
```

Reply:
done